include/envoy/router/router.h

```diff
@@ -727,6 +727,17 @@ class RouteEntry : public ResponseEntry {
    */
   virtual const CorsPolicy* corsPolicy() const PURE;
 
+  /**
+   * Returns the URL path as it will be calculated by finalizeRequestHeaders
+   * using current values of headers. Note that final path may be different if
+   * headers change before finalization.
+   * @param headers supplies the request headers.
+   * @return absl::optional<std::string> the value of the URL path after rewrite or an empty
+   *         value if rewrite is not configured.
```

Comment:
empty value -> absl::nullopt (to clarify not empty string)

Reply:
Done